### PR TITLE
Project details

### PR DESCRIPTION
**Pull request by DeeaMatei**
#### [Task](task_url)
https://trello.com/c/B1GUAOXw/172-implement-projects-details-screen

### CODE REVIEW

**Comment by DeeaMatei** [comment]
[Implement projects details screen](https://trello.com/c/B1GUAOXw/172-implement-projects-details-screen)

File: features/projects/build.gradle.kts [hunk 1]
```diff
@@ -21,4 +21,5 @@ apollo {
 
 dependencies {
     implementation(project(BuildModules.Commons.UI))
+    testImplementation(project(mapOf("path" to ":features:freaks")))
```
Comment (AlexandraDamaschin) [inline features/projects/build.gradle.kts]: Why do we need this?

File: features/projects/src/main/graphql/com.agilefreaks.freaks_catalog.features.projects/ProjectDetails.graphql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+query ProjectDetails {
```
Comment (AlexandraDamaschin) [inline features/projects/src/main/graphql/com.agilefreaks.freaks_catalog.features.projects/ProjectDetails.graphql]: Is this a ProjectDetails only?

File: features/projects/src/main/kotlin/com/agilefreaks/freaks_catalog/features/projects/ProjectsApolloClient.kt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.agilefreaks.freaks_catalog.features.projects
```
Comment (AlexandraDamaschin) [inline features/projects/src/main/kotlin/com/agilefreaks/freaks_catalog/features/projects/ProjectsApolloClient.kt]: From what I remember this is added in the project, why do we need to create another apolloclient?

File: features/projects/src/test/kotlin/com/agilefreaks/freaks_catalog/features/projects/CoroutineRule.kt [hunk 4]
```diff
@@ -0,0 +1,25 @@
+package com.agilefreaks.freaks_catalog.features.projects
```
Comment (AlexandraDamaschin) [inline features/projects/src/test/kotlin/com/agilefreaks/freaks_catalog/features/projects/CoroutineRule.kt]: This seems to be copy/paste code? I have a deja vu I've seen this code already in our app

File: features/projects/src/test/kotlin/com/agilefreaks/freaks_catalog/features/projects/ProjectDetailsViewModelTest.kt [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+@ExperimentalCoroutinesApi
+class ProjectDetailsViewModelTest {
+    @get: Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    private val projectDetailsRepository = mock<ProjectDetailsRepository>()
+
+    @ExperimentalCoroutinesApi
+    @Test
+    fun `loadProject will populate project with values from repository`() = runBlockingTest {
```
Comment (AlexandraDamaschin) [inline features/projects/src/test/kotlin/com/agilefreaks/freaks_catalog/features/projects/ProjectDetailsViewModelTest.kt]: Why are we testing this?

**Review by AlexandraDamaschin — CHANGES_REQUESTED** [review]
Some small questions/improvements

File: features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/Freak.kt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.agilefreaks.freaks_catalog.features.freaks
+
+data class Freak(
+    val id:String,
+    val firstName: String,
+    val photo: String
+)
```
Comment (cipry13) [inline features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/Freak.kt]: why do you need this?

**Comment by cipry13** [comment]
Project description is missing

**Comment by cipry13** [comment]
Photo doesn't have rounded corners and border

File: features/projects/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -1,3 +1,7 @@
 <resources>
     <string name="projects_title">PROJECTS</string>
+    <string name="picture">Picture</string>
+    <string name="skills">Skills: %1$s</string>
+    <string name="description">Description: %1$s</string>
```
Comment (cipry13) [inline features/projects/src/main/res/values/strings.xml]: "Description:" should not appear on screen

File: features/projects/src/main/kotlin/com/agilefreaks/freaks_catalog/features/projects/details/ProjectDetailsDataSource.kt [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+import com.apollographql.apollo.coroutines.await
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+interface ProjectDetailsDataSource {
+    suspend fun getProjectDetails(id: String): ProjectDetailsQuery.Data?
+}
+
+class ProjectDetailsApolloDataSource : ProjectDetailsDataSource {
+    override suspend fun getProjectDetails(id: String): ProjectDetailsQuery.Data? =
```
Comment (cipry13) [inline features/projects/src/main/kotlin/com/agilefreaks/freaks_catalog/features/projects/details/ProjectDetailsDataSource.kt]: whose id is this?

File: features/projects/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -1,3 +1,7 @@
 <resources>
     <string name="projects_title">PROJECTS</string>
+    <string name="picture">Picture</string>
+    <string name="skills">Skills: %1$s</string>
```
Comment (cipry13) [inline features/projects/src/main/res/values/strings.xml]: Skill should be renamed to Technologies

**Review by srobert99 — APPROVED** [review]

File: features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/details/DetailsFragmentModule.kt [hunk 10]
```diff
@@ -3,7 +3,7 @@ package com.agilefreaks.freaks_catalog.features.freaks.details
 import org.koin.androidx.viewmodel.dsl.viewModel
 import org.koin.dsl.module
 
-val detailsFragmentModule = module {
+val projectDetailsFragmentModule = module {
```
Comment (deniscoman) [inline features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/details/DetailsFragmentModule.kt]: Something doesn't feel right here 🤔

File: features/freaks/src/test/kotlin/com/agilefreaks/freaks_catalog/features/freaks/DetailsViewModelTest.kt [hunk 11]
```diff
@@ -1,26 +1,23 @@
...
-import kotlinx.coroutines.test.runBlockingTest
 import org.junit.Rule
 import org.junit.Test
 
-@ExperimentalCoroutinesApi
 class DetailsViewModelTest {
+    @ExperimentalCoroutinesApi
     @get:Rule
-    var instantTaskExecutorRule = InstantTaskExecutorRule()
+    var coroutineRule = CoroutineRule()
```
Comment (deniscoman) [inline features/freaks/src/test/kotlin/com/agilefreaks/freaks_catalog/features/freaks/DetailsViewModelTest.kt]: It feels like that on rebase you keep the wrong version here :(

File: features/projects/src/main/kotlin/com/agilefreaks/freaks_catalog/features/projects/details/ProjectDetailsFragmentModule.kt [hunk 12]
```diff
@@ -0,0 +1,12 @@
+package com.agilefreaks.freaks_catalog.features.projects.details
+
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.dsl.module
+
+val detailsFragmentModule = module {
```
Comment (deniscoman) [inline features/projects/src/main/kotlin/com/agilefreaks/freaks_catalog/features/projects/details/ProjectDetailsFragmentModule.kt:6]: It's not only here the problem, but this variable and this file name should'n contain fragment in it. It doesn't have anything to do with fragments

**Review by deniscoman — CHANGES_REQUESTED** [review]

**Review by AlexandraDamaschin — APPROVED** [review]
Does this improve software quality? Yes

**Pull request merged**